Makefile.pre.in

```diff
@@ -440,7 +440,15 @@ DTRACE_DEPS = \
 
 # Default target
 all:		@DEF_MAKE_ALL_RULE@
-build_all:	$(BUILDPYTHON) oldsharedmods sharedmods gdbhooks Programs/_testembed python-config
+build_all:	check-clean-src $(BUILDPYTHON) oldsharedmods sharedmods gdbhooks \
```

Comment:
I would prefer to put check-clean-src in $(BUILDPYTHON). I tested, it works:

```
# Build the interpreter
$(BUILDPYTHON):	check-clean-src Programs/python.o $(LIBRARY) $(LDLIBRARY) $(PY3LIBRARY)
	$(LINKCC) $(PY_LDFLAGS) $(LINKFORSHARED) -o $@ Programs/python.o $(BLDLIBRARY) $(LIBS) $(MODLIBS) $(SYSLIBS) $(LDLAST)
```

Reply:
1. One problem is that in this case the $(BUILDPYTHON) target is rebuilt even when it is up to date, for the following reason:
```
A phony target should not be a prerequisite of a real target file;
if it is, its recipe will be run every time make goes to update that file.
```
This is a quote from [Phony Targets](https://www.gnu.org/software/make/manual/make.html#Phony-Targets).

2. Another minor problem is that the recipe of check-clean-src may be run multiple times.

Reply:
Oh oh, I didn't know.